File: java-sdk/radar-catalog-server/src/main/java/org/radarbase/schema/service/SourceCatalogueJerseyEnhancer.java
package org.radarbase.schema.service;

import jakarta.inject.Singleton;
import org.glassfish.jersey.internal.inject.AbstractBinder;
import org.glassfish.jersey.server.ResourceConfig;
import org.jetbrains.annotations.NotNull;
import org.radarbase.jersey.config.ConfigLoader.Filters;
import org.radarbase.jersey.config.JerseyResourceEnhancer;
import org.radarbase.schema.specification.SourceCatalogue;

public class SourceCatalogueJerseyEnhancer implements JerseyResourceEnhancer {
    private final SourceCatalogue sourceCatalogue;

    public SourceCatalogueJerseyEnhancer(SourceCatalogue sourceCatalogue) {
        this.sourceCatalogue = sourceCatalogue;
    }

    @NotNull
    @Override
    public Class<?>[] getClasses() {
        return new Class[] {
            Filters.INSTANCE.getLogResponse(),
            SourceCatalogueService.class,
        };
    }

    @NotNull
    @Override
    public String[] getPackages() {
        return new String[0];
    }

    @Override
    public void enhanceBinder(
            @NotNull AbstractBinder abstractBinder) {
        enhance(abstractBinder);
    }

    @Override
    public void enhanceResources(@NotNull ResourceConfig resourceConfig) {
        enhance(resourceConfig);
    }

    @Override
    public void enhance(@NotNull AbstractBinder abstractBinder) {
        abstractBinder.bindFactory(() -> sourceCatalogue)
                .to(SourceCatalogue.class)
                .in(Singleton.class);
    }

    @Override
    public void enhance(@NotNull ResourceConfig resourceConfig) {
        // do nothing
    }
}

File: java-sdk/radar-catalog-server/src/main/java/org/radarbase/schema/service/SourceCatalogueJerseyEnhancer.kt
package org.radarbase.schema.service

import jakarta.inject.Singleton
import org.glassfish.jersey.internal.inject.AbstractBinder
import org.radarbase.jersey.enhancer.JerseyResourceEnhancer
import org.radarbase.jersey.filter.Filters.logResponse
import org.radarbase.schema.specification.SourceCatalogue

class SourceCatalogueJerseyEnhancer(
    private val sourceCatalogue: SourceCatalogue,
) : JerseyResourceEnhancer {
    override val classes: Array<Class<*>> = arrayOf(
        logResponse,
        SourceCatalogueService::class.java,
    )

    override val packages: Array<String> = emptyArray()

    override fun AbstractBinder.enhance() {
        bindFactory { sourceCatalogue }
            .to(SourceCatalogue::class.java)
            .`in`(Singleton::class.java)
    }
}
